crates/hir-def/src/expander.rs

```diff
@@ -49,6 +49,10 @@ impl Expander {
         }
     }
 
+    pub(crate) fn span_map(&self) -> Option<&SpanMap> {
+        self.span_map.get()
+    }
```

Comment:
This seems a bit weird of an API, this shouldnt be optional (it only is due to not always being initialized given the lazy OnceCell)

Reply:
We also need to be careful with the span map as that currently accessing it unfortunately completely breaks incrementality :( Though it probably is fine here (as bodies are invalidated anyways), and type inference only when its actually using macros then (assuming we only access it then)

Reply:
I only need this after expanding a macro (to access the new span map), and a span map is always available in this case, so it doesn't really matter if it returns an `Option` or initializes the cell. At first I tried making `enter_expand()` return the span map too, but this is a more intrusive change (even if it saves an unwrap).

Also inference can't access this since this is `pub(crate)`.